vulnfeeds/cves/versions_test.go

```diff
@@ -374,3 +374,112 @@ func TestExtractGitCommit(t *testing.T) {
 		}
 	}
 }
+
+func TestNormalize(t *testing.T) {
+	tests := []struct {
+		description               string
+		inputVersion              string
+		expectedNormalizedVersion string
+		expectedOk                bool
+	}{
+		{
+			description:               "Empty string",
+			inputVersion:              "",
+			expectedNormalizedVersion: "",
+			expectedOk:                false,
+		},
+		{
+			description:               "Garbage version",
+			inputVersion:              "hjlk;gfdhjkgf",
+			expectedNormalizedVersion: "",
+			expectedOk:                false,
+		},
+		{
+			description:               "Valid supported version #1",
+			inputVersion:              "1.0",
+			expectedNormalizedVersion: "1-0",
+			expectedOk:                true,
+		},
+		{
+			description:               "Valid supported version #2",
+			inputVersion:              "22.3rc1",
+			expectedNormalizedVersion: "22-3-rc1",
+			expectedOk:                true,
+		},
+		{
+			description:               "Valid supported version #3",
+			inputVersion:              "1.2.3.4.5-rc1",
+			expectedNormalizedVersion: "1-2-3-4-5-rc1",
+			expectedOk:                true,
+		},
+		{
+			description:               "Valid supported version #4",
+			inputVersion:              ".1",
+			expectedNormalizedVersion: "1",
```

Comment:
I wonder if this special case should be 0-1 instead. (the bug.py implementation may be wrong here). 

This is also more consistent with https://maven.apache.org/pom.html#version-order-specification where ".1" actually means "0.1"